CMakeLists.txt

```diff
@@ -275,6 +283,57 @@ if (DEFINED HAVE_INET_PTON)
 	add_definitions(-DHAVE_INET_PTON=1)
 endif()
 
+# Check for pthread_getname_np(3) and pthread_setname_np(3)
+#    used in srtcore/threadname.h.
+# Some BSD distros need to include <pthread_np.h> for pthread_getname_np().
+# TODO: Some BSD distros have pthread_get_name_np() and pthread_set_name_np()
+#     instead of pthread_getname_np() and pthread_setname_np().
+set(CMAKE_REQUIRED_DEFINITIONS
+   -D_GNU_SOURCE -D_DARWIN_C_SOURCE -D_POSIX_SOURCE=1)
+set(CMAKE_REQUIRED_FLAGS "-pthread")
+message(STATUS "Checking for pthread_(g/s)etname_np in 'pthread_np.h':")
+unset(HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H)
+unset(HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H CACHE)
+check_symbol_exists(
+   pthread_getname_np "pthread_np.h" HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H)
+if (HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H)
+   add_definitions(-DHAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H=1)
+endif()
+unset(HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H)
+unset(HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H CACHE)
+check_symbol_exists(
+   pthread_setname_np "pthread_np.h" HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H)
+if (HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H)
+   add_definitions(-DHAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H=1)
+endif()
+unset(HAVE_PTHREAD_GETNAME_NP)
+unset(HAVE_PTHREAD_GETNAME_NP CACHE)
+message(STATUS "Checking for pthread_(g/s)etname_np in 'pthread.h':")
+check_symbol_exists(pthread_getname_np "pthread.h" HAVE_PTHREAD_GETNAME_NP)
+if (HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H)
+   set(HAVE_PTHREAD_GETNAME_NP TRUE)
+endif()
+unset(HAVE_PTHREAD_SETNAME_NP)
+unset(HAVE_PTHREAD_SETNAME_NP CACHE)
+check_symbol_exists(pthread_setname_np "pthread.h" HAVE_PTHREAD_SETNAME_NP)
+if (HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H)
+   set(HAVE_PTHREAD_SETNAME_NP TRUE)
+endif()
+if (HAVE_PTHREAD_GETNAME_NP)
+   add_definitions(-DHAVE_PTHREAD_GETNAME_NP=1)
+endif()
+if (HAVE_PTHREAD_SETNAME_NP)
+   add_definitions(-DHAVE_PTHREAD_SETNAME_NP=1)
+endif()
```

Comment:
This code is better to be moved in a dedicated CMake function in `scripts/haiUtil.cmake` or a separate dedicated file.

Please use tabs instead of space in CMake files.

Also, I suppose it must be run only if the target system is not Windows (`if (NOT MICROSOFT)`). At least checking with `pthreads` port for Windows non is found:
```
HAVE_PTHREAD_GETNAME_NP_IN_PTHREAD_NP_H=
HAVE_PTHREAD_SETNAME_NP_IN_PTHREAD_NP_H=
HAVE_PTHREAD_GETNAME_NP=
HAVE_PTHREAD_SETNAME_NP=
```

Reply:
Note that `MICROSOFT` symbol is - in simplest explanation - understood as "Windows with Microsoft Visual Studio". It distinguishes to two other possible compiling options on Windows platform - MinGW32 and Cygwin.

Reply:
BTW why do these symbols have separate GETNAME and SETNAME versions? Shouldn't it state that we should simply have two check-entries: 1) Do we have pthread_get/setname_np available, 2) Do we need extra header for it.

Reply:
Some versions of BSD require `#include <pthread_np.h>` others have it in '#include <pthread.h>`. Some runtimes have only `pthread_setname_np()` and not `pthread_getname_np()`. It is all explained in the comments I added to `srtcore/threadname.h`.

Also these functions can be available when using Visual Studio when building with POSIX Pthreads. They are part of the PThreads library.

Reply:
Yeah, but then you either have this functionality or not. What good is ability to set the name if you can't read them? Reading is required for two cases: in the logs and potentially in gdb (including coredumps). Not sure whether on the systems that allow for setting, but not for getting, gdb has any "secret" method to read these names, although on one platform - not sure which one, but it might be Linux on some weird device - I had names visible in the logs, but not in gdb.

Reply:
I am merely fixing the build in the case where one of the 4 options are true, both functions exist, one of the two functions exist, neither exist, or one or both exist but need to include a different header file. Setting it still may be useful for debugging purposes either with `GDB` or `PS` which is why I think that when only one of the functions is available without the other, it tends to be the case that `pthread_setname_np()` is present but `pthread_getname_np()` is not. For instance in the current `MUSL` runtime.